spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreReactiveOperations.java

```diff
@@ -129,6 +129,24 @@
 	 */
 	<T> Mono<Void> deleteById(Publisher<String> idPublisher, Class entityClass);
 
+  /**
+   * Executes a query represented as query builder and returns results of the given domain type.
+   * @param <T> the type param of the domain type.
+   * @param builder the query builder.
+   * @param entityClass the domain type of entities.
+   * @return {@link Flux} emitting the found entities.
+   * @since 1.2.4
+   */
+  <T> Flux<T> execute(StructuredQuery.Builder builder, Class<T> entityClass);
+
+  /**
+   * Creates FirestoreReactiveOperations object with a provided parent.
+   * The parent doesn't have to exist in Firestore, but should have a non-empty id field.
```

Comment:
This javadoc needs a little bit more description; I think you have to define "parent" more clearly.

There needs to be something that says that by default the template operates at the root level collection `/` and then once you pass an entity in here, a new template is created with that entity at the root collection and that all operations/queries will be scoped to that document's subcollections.

Also need to clarify what happens if you pass in an entity in here that is not a child of the current parent with `withParent(..)` - does it still work?

Reply:
The "scope" actually explains it. When you save any entity, it is saved within the scope of this parent's subcollections.